src/lib/src/viewer/viewer.component.ts

```diff
@@ -113,10 +115,19 @@ export class ViewerComponent implements OnInit, OnDestroy, OnChanges {
   }
 
   toggleToolbarsState(mode: ViewerMode): void {
-    if (mode === ViewerMode.DASHBOARD) {
-      this.header.state = this.footer.state = 'show';
-    } else if (mode === ViewerMode.PAGE) {
-      this.header.state = this.footer.state = 'hide';
+    switch (mode) {
+      case ViewerMode.DASHBOARD:
+        this.header.state = this.footer.state = 'show';
+        if (this.config.navigationControlEnabled && this.osdToolbar) {
```

Comment:
Can you add a comment about the last "&& this.osdToolbar"? As you told me, it's caused by the ViewChild not being initialized at the startup of the app because of the ngIf on the child. Not sure what to write though. Maybe link to a relevant stackoverflow discussion?

Reply:
The viewchild (this.osdToolbar) is undefined initially, presumably because the ngIf in the template is not evaluated before toggleToolbarState is executed. Removing ngIf from the template doesn't cause this behavior, but then the element is rendered in the DOM and hidden by the animation-state. We could possibly use some lifecycles-hooks or detectChanges() in some nifty way if this solution is not satisfactory :)

Reply:
I was thinking of a code comment to remember this information when we later wonder why we did it this way. Sorry for not being clear about that.